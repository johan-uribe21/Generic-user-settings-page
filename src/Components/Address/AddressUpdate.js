import React from 'react';

// Controlled display Component, state held in App.js
const AddressUpdate = ({address, onChange, onSubmit}) => {

  return (
    <form
      className="beam-text"
      onSubmit={onSubmit}
    >
      <label className='pa2'>Street: <br/>
        <input 
        className="beam-text pa2 input-reset b--dotted bg-transparent w-100 measure"
        id="street" 
        type="text" 
        size="40"
        onChange={onChange}
        value = {address.street} // for controlled component, state controls values
        />
      </label><br/>
      <label className='pa2'>City: <br/>
        <input 
        className="beam-text pa2 input-reset b--dotted bg-transparent w-100 measure"
        id="city" 
        type="text" 
        size="40"
        onChange={onChange}
        value = {address.city} // for controlled component, state controls values
        />
      </label><br/>
      <label className='pa2'>State: <br/>
        <input 
        className="beam-text pa2 input-reset b--dotted bg-transparent w-100 measure"
        id="state" 
        type="text" 
        size="40"
        onChange={onChange}
        value = {address.state} // for controlled component, state controls values
        />
      </label><br/>
      <label className='pa2'>Zipcode: <br/>
        <input 
        className="beam-text pa2 input-reset b--dotted bg-transparent w-100 measure"
        id="zipcode" 
        type="text" 
        size="40"
        onChange={onChange}
        value = {address.zipcode} // for controlled component, state controls values
        />
      </label><br/>
      <input 
          type="submit"
          className='beam-edit-button' 
          value='Update'
        />
    </form>
  )
}

export default AddressUpdate;