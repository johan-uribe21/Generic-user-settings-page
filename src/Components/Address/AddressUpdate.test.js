import {shallow, configure} from 'enzyme';
import AddressUpdate from './AddressUpdate';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() }); 

const mockAddress =  {
    street: '123 elm',
    city: 'cbus',
    state: 'OH',
    zipcode: '43023'
}
const event = {
  preventDefault() {},
  target: {value: 'the-value'}
}
const mockOnChange = jest.fn();
const mockOnSubmit = jest.fn();
const wrapper= shallow(
  <AddressUpdate 
    address={mockAddress} 
    onChange={mockOnChange} 
    onSubmit={mockOnSubmit}/>);
  
describe('tests snapshot, event handdling, and submission logic', () => {
  it('renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('calls onChange function when the user types into the form', () => {
    // simulates the onchange event with a synthetic event
    wrapper.find('#street').simulate('change', event);
    expect(mockOnChange).toBeCalledWith(event);
  });

});
  