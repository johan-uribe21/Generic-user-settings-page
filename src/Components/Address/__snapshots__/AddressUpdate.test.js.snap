// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests snapshot, event handdling, and submission logic renders the component 1`] = `ShallowWrapper {}`;
